### ISSUE TITLE
Bruteforce index layer cannot be served through TF Serving docker container!

### ISSUE THREAD

**Issue by siyu1992** [tensorflow/recommenders#499]
Hi, I have implemented a application based on [https://www.tensorflow.org/recommenders/examples/deep_recommenders](url), and I found that the model cannot be served through TF Serving docker container while it worked fine by  tf.saved_model.load() method. 

So, I'm comfused about this. 

request:
`curl -d '{"instances": [{"msisdn": ["13876539023"],"prov_cd": ["gd"],"reg_cnl": [9],"usr_brd": [1],"usr_star_lvl": [6]}]}' -X POST http://localhost:8501/v1/models/simple_dssm_bruce:predict`

error:
{
    "error": "ConcatOp : Dimension 2 in both shapes must be equal: shape[0] = [1,1,64] vs. shape[1] = [1,1,4]\n\t [[{{node brute_force/query_model/user_model/concat}}]]"
}

user_model  is implemented as follows:

`class UserModel(tf.keras.Model):

    def __init__(self,embedding_dim):
        super().__init__()
        self.embedding_dim = embedding_dim

        self.user_embedding = tf.keras.Sequential([
        tf.keras.layers.StringLookup(
            vocabulary=unique_msisdns, mask_token=None),
        tf.keras.layers.Embedding(len(unique_msisdns) + 1, self.embedding_dim),
    ])
       
        self.normalized_reg_cnl = tf.keras.layers.Normalization(
        axis=None
    )

        self.normalized_reg_cnl.adapt(reg_cnl)
        
        self.normalized_usr_brd = tf.keras.layers.Normalization(
        axis=None
    )

        self.normalized_usr_brd.adapt(usr_brd)
        
        self.normalized_usr_star_lvl = tf.keras.layers.Normalization(
        axis=None
    )

        self.normalized_usr_star_lvl.adapt(usr_star_lvl)    
        
        #embedding feature
        self.prov_cd_embedding = tf.keras.Sequential([
        tf.keras.layers.StringLookup(
            vocabulary=unique_prov_cd, mask_token=None),
        tf.keras.layers.Embedding(len(unique_prov_cd) + 1, 4),
    ])     

    def call(self, inputs):
        # Take the input dictionary, pass it through each input layer,
        # and concatenate the result.
        return tf.concat([
        self.user_embedding(inputs["msisdn"]),
        self.prov_cd_embedding(inputs["prov_cd"]),
        tf.reshape(self.normalized_reg_cnl(inputs["reg_cnl"]), (-1, 1)),
        tf.reshape(self.normalized_usr_brd(inputs["usr_brd"]), (-1, 1)),
        tf.reshape(self.normalized_usr_star_lvl(inputs["usr_star_lvl"]), (-1, 1)),
        ], axis=1)`

Could you please take a look at this error, it didn't come out while using  tf.saved_model.load(model_path) to make a prediction.

Thanks a lot!
